modules/container_deployment/main.tf

```diff
@@ -71,14 +67,38 @@ resource "helm_release" "jaeger-operator" {
 
   repository = "https://jaegertracing.github.io/helm-charts"
   chart      = "jaeger-operator"
+  values = [
+    file("${path.module}/jaeger_values.yaml")
+  ]
+}
 
-  set {
-    name  = "jaeger.create"
-    value = "true"
+# Basically copied from Hono Helm charts
+resource "kubernetes_config_map" "grafana_hono_dashboards" {
+  metadata {
+    name    = "grafana-hono-dashboards"
+    labels  = {
+      "grafana_dashboard": "1" # Add labels so that Grafana finds these
+    }
   }
 
-  set {
-    name  = "metadata.name"
-    value = "simple"
+  data = {
+    "jvm-details.json"      = file("${path.module}/Grafana_Dashboards/JVM_details_grafana_ds.json")
+    "message-details.json"  = file("${path.module}/Grafana_Dashboards/Message_details_grafana_ds.json")
+    "overview.json"         = file("${path.module}/Grafana_Dashboards/Overview_grafana_ds.json")
```

Comment:
Can you elaborate a bit how one adds new Grafana dashboards and/or new Grafana datasources?
I quickly skimmed through https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack and saw references to jsonnet configuration used in https://github.com/prometheus-operator/kube-prometheus and to https://github.com/grafana/helm-charts/tree/main/charts/grafana.
There is a https://github.com/grafana/helm-charts/tree/main/charts/grafana#import-dashboards so it might be worth mentioning that this mechanism uses https://github.com/grafana/helm-charts/tree/main/charts/grafana#sidecar-for-dashboards.

Reply:
If you already have a Grafana dashboard json file, one way is to create a kubernetes configmap that stores those jsons and give that configmap label 'grafana_dashboard'. Then the sidecar container looks for the right configmaps.

I didn't yet look at adding datasources, because kube-prometheus-stack adds Prometheus as a datasource by default, but I suspect that's the easier part. Defining new dashboard might be more challenging.